### PR TITLE
speed up platform probe 50ms -> 5ms by avoiding phobos imports

### PR DESCRIPTION
**Pull request by MartinNowak**

### CODE REVIEW

**Comment by MartinNowak** [comment]
Should this go into stable as it fixes a performance regression?

File: source/dub/compilers/utils.d [hunk 1]
```diff
@@ -248,39 +248,40 @@ NativePath generatePlatformProbeFile()
...
-		`	return '"' ~ ret.data.join("\", \"") ~ "\", "; }` ~ q{
-
-		string determineCompiler() } ~ '{' ~ compilerCheck ~ "	}");
+		string[] determinePlatform() } ~ '{' ~ platformCheck ~ '}' ~ q{
+		string[] determineArchitecture() } ~ '{' ~ archCheck ~ '}' ~ q{
+		string determineCompiler() } ~ '{' ~ compilerCheck ~ '}';
 
+	auto path = getTempFile("dub_platform_probe", ".d");
+	auto fil = openFile(path, FileMode.createTrunc);
+	scope (failure) {
```
Comment (wilzbach) [inline source/dub/compilers/utils.d]: Why not `scope(exit)` and remove `fil.close()` in L285?

**Comment by wilzbach** [comment]
> Should this go into stable as it fixes a performance regression introduced by 61cf37f?

Sounds very reasonable.

File: source/dub/compilers/utils.d (hunk 1 above)
Comment (s-ludwig) [reply source/dub/compilers/utils.d]: `File` should be ref counted anyway, so neither should be neccessary.

**Comment by s-ludwig** [comment]
From my side too this can go into stable, although labeling this as a "regression" is maybe a bit too much when talking about 45ms ;)

**Pull request merged**